Refresh.GameServer/Services/CommandService.cs

```diff
@@ -0,0 +1,103 @@
+using Bunkum.HttpServer;
+using Bunkum.HttpServer.Services;
+using JetBrains.Annotations;
+using MongoDB.Bson;
+using NotEnoughLogs;
+using Refresh.GameServer.Database;
+using Refresh.GameServer.Types.Commands;
+using Refresh.GameServer.Types.UserData;
+
+namespace Refresh.GameServer.Services;
+
+public class CommandService : EndpointService
+{
+    private readonly MatchService _match;
+    
+    public CommandService(LoggerContainer<BunkumContext> logger, MatchService match) : base(logger) {
+        this._match = match;
+    }
+
+    private readonly HashSet<ObjectId> _usersPublishing = new();
+
+    /// <summary>
+    /// Start tracking the user, eg. they started publishing
+    /// </summary>
+    /// <param name="id">The user ID</param>
+    public void StartPublishing(ObjectId id)
+    {
+        //Unconditionally add the user to the set
+        _ = this._usersPublishing.Add(id);
+    }
+
+    /// <summary>
+    /// Stop tracking the user, eg. they stopped publishing
+    /// </summary>
+    /// <param name="id">The user ID</param>
+    public void StopPublishing(ObjectId id)
+    {
+        //Unconditionally remove the user from the set
+        _ = this._usersPublishing.Remove(id);
+    }
+
+    public bool IsPublishing(ObjectId id) => this._usersPublishing.Contains(id);
+
+    /// <summary>
+    /// Parse a command string into a command object
+    /// </summary>
+    /// <param name="str">Command string</param>
+    /// <returns>Parsed command</returns>
+    /// <exception cref="FormatException">When the command is in an invalid format</exception>
+    [Pure]
+    public Command ParseCommand(string str)
```

Comment:
I know we talked aboutthis last night, but the fact that this could definitely be using Spans for this function to make it allocation-less still irks me. I can do this for you if you'd like